docs/source/howto/installation.rst

```diff
@@ -26,23 +26,271 @@ Configuring your installation
+       you are responsible for this at the moment.
+       This can be achieved for instance by submitting only a maximum number of workflows to AiiDA, and submitting new ones only when the previous ones complete.
+
+     * Tune the parameters that AiiDA uses to avoid overloading the supercomputer with connections or batch requests.
+       For SSH transports, the default is 30 seconds, which means that when each worker opens a SSH connection to a computer, it will reuse it as long as there are tasks to execute and then close it.
+       Opening a new connection will not happen before 30 seconds has passed from the opening of the previous one.
+
+       We stress that this is *per daemon worker*, so that if you have 10 workers, your supercomputer will on average see 10 connections every 30 seconds.
+       Therefore, if you are using many workers and you mostly have long-running jobs, you can set a longer time (e.g., 120 seconds) by reconfiguring the computer with ``verdi computer configure ssh <COMPUTER_NAME>`` and changing the value
+       of the *Connection cooldown time* or, alternatively, by running::
+
+         verdi computer configure ssh --non-interactive --safe-interval <SECONDS> <COMPUTER_NAME>
+
+     * In addition to the connection cooldown time described above, AiiDA also avoids running too often the command to get the list of jobs in the queue (``squeue``, ``qstat``, ...), as this can also impact the performance of the scheduler.
+       For a given computer, you can increase how many seconds must pass between requests.
+       First load the computer in a shell with ``computer = load_computer(<COMPUTER_NAME>)``.
+       You can check the current value in seconds (by default, 10) with ``computer.get_minimum_job_poll_interval()``.
+       You can then set it to a higher value using::
+
+         computer.set_minimum_job_poll_interval(<NEW_VALUE_SECONDS>)
 
 .. _how-to:installation:update:
 
 Updating your installation
 ==========================
 
-`#4003`_
+1. Enter the python environment where AiiDA is installed
+2. Finish all running calculations.
+   After migrating your database, you will not be able to resume unfinished calculations.
+   Data of finished calculations will of course be automatically migrated.
+3. Stop the daemon using ``verdi daemon stop``.
+4. :ref:`Create a backup of your database and repository<how-to:installation:backup>`.
+
+   .. warning::
+
+      Once you have migrated your database, you can no longer go back to an older version of ``aiida-core`` (unless you restore your database and repository from a backup).
+
+5. Update your ``aiida-core`` installation.
+
+    * If you have installed AiiDA through ``pip`` simply run: ``pip install --upgrade aiida-core``.
+    * If you have installed from the git repository using ``pip install -e .``, first delete all the ``.pyc`` files (``find . -name "*.pyc" -delete``) before updating your branch.
```

Comment:
This has caused a lot of pain in the past (@sphuber can confirm), so I would vote for keeping this (unless we are sure that in python 3 the problem is solved, but I'm not sure it is). If you feel we should mention `__pycache__`, it's ok with me (should we just delete those as well?). But I think that if we delete the *.pyc, in py3 these are inside the `__pycache__` folder and the suggestion would still work.

In detail, the reason for the problems that you get if you don't delete them is when you delete or rename a file. In particular, this happened a lot in the 0.x->1.x transition (might be less relevant in the 1.x versions, but we might still end up doing some refactoring of internal python files). However, since *.pyc are not tracked by git, git would leave them at their old location together with their folders, and python would happily continue to load them.